buildbot-conf/cordova.conf

```diff
@@ -234,8 +245,15 @@ plugins_cleanup_steps = [
 common_plugins_steps = plugins_cleanup_steps + get_medic_steps + [
 
     # download medic's config to slave
-    Download(mastersrc=PROJECTS_CONFIG_FILE, slavedest='cordova-medic/cordova-repos.json'),
-    Download(mastersrc=MEDIC_CONFIG_FILE, slavedest='cordova-medic/config.json'),
+    Download(mastersrc=MEDIC_CONFIG_FILE, slavedest='cordova-medic/config.json', description='downloading master\'s config'),
+
+    # download repo config
+    # NOTE:
+    #      only one of these steps should be executed; thanks
+    #      to Buildbot there is no good if-else construct for
+    #      builds, so we have two steps with 'doStepIf's
+    SH(command=['curl', P(REPOS_PROPERTY_NAME), '--output', 'cordova-medic/cordova-repos.json'], description='downloading custom repo config', doStepIf=dont_use_default_repos),
+    Download(mastersrc=PROJECTS_CONFIG_FILE, slavedest='cordova-medic/cordova-repos.json', description='downloading default repo config', doStepIf=use_default_repos),
```

Comment:
I wish it worked that way, but sadly [it does not](http://docs.buildbot.net/current/manual/cfg-properties.html#using-properties-in-steps). :disappointed: The way to do what you're talking about is to implement a [custom step that runs shell commands](http://docs.buildbot.net/current/manual/customization.html#running-commands), which is just as convoluted.

The way to solve it properly is to have only one download step and use a *default* property value, like so: 

    `P('repositories_config', default=PROJECTS_CONFIG_FILE)`

The issue right now is that the default file uses `FileDownload`, while the custom file uses `ShellCommand(command=['curl', ...])`, and making both use the same one is also a convoluted process. This will be fixed entirely once we switch to using [Buildbot Codebases](http://docs.buildbot.net/current/manual/concepts.html#multiple-codebase-builds) for managing multiple repositories.